sql/driver/wire.go

```diff
@@ -80,14 +100,17 @@ func (d Datum) Value() (driver.Value, error) {
 		val = t.BytesVal
 	case *Datum_StringVal:
 		val = t.StringVal
+	case *Datum_DateVal:
+		val = Date{t.DateVal.GoTime()}
```

Comment:
https://github.com/cockroachdb/cockroach/blob/af72b76c1dff094cfe667f8b1b124081f7ba6452/sql/parser/eval.go#L1067

If the server didn't truncate dates then `==` and other operators for dates would be broken.